pkg/pod/script.go

```diff
@@ -107,22 +108,18 @@ func convertListOfSteps(steps []v1beta1.Step, initContainer *corev1.Container, p
 		// non-nil init container.
 		*placeScripts = true
 
+		script = encodeScript(script)
+
 		// Append to the place-scripts script to place the
 		// script file in a known location in the scripts volume.
 		tmpFile := filepath.Join(scriptsDir, names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("%s-%d", namePrefix, i)))
-		// heredoc is the "here document" placeholder string
-		// used to cat script contents into the file. Typically
-		// this is the string "EOF" but if this value were
-		// "EOF" it would prevent users from including the
-		// string "EOF" in their own scripts. Instead we
-		// randomly generate a string to (hopefully) prevent
-		// collisions.
-		heredoc := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("%s-heredoc-randomly-generated", namePrefix))
+		heredoc := "_EOF_" // underscores because base64 doesnt include them in its alphabet
 		initContainer.Args[1] += fmt.Sprintf(`tmpfile="%s"
 touch ${tmpfile} && chmod +x ${tmpfile}
 cat > ${tmpfile} << '%s'
 %s
 %s
+/tekton/tools/entrypoint decode-script "${tmpfile}"
```

Comment:
NIT: can we use something more meaningful here instead of `tmpfile`, may be a script file or a encoded script file? looks odd to see a `tmpfile` in pod description:

```
kubectl get pod/dollar-dollar-taskrun-lztnb-pod-tvs2h -o json |  jq .spec.initContainers
...
{
    "args": [
      "-c",
      "tmpfile=\"/tekton/scripts/script-0-ck986\"\ntouch ${tmpfile} && chmod +x ${tmpfile}\ncat > ${tmpfile} << '_EOF_'\nIyEvYmluL3NoCnNldCAteGUKZWNobyAndHdvIGRvbGxhciBzaWduczogJCQnCmVjaG8gJ2ZvdXIgZG9sbGFyIHNpZ25zOiAkJCQkJwplY2hvICd0d28gZG9sbGFyIHNpZ25zIGZyb20gYW4gZW52aXJvbm1lbnQgdmFyaWFibGU6ICcgJERPTExBUl9ET0xMQVJfRU5WCg==\n_EOF_\n/tekton/tools/entrypoint decode-script \"${tmpfile}\"\n"
...
```

Reply:
good call, created https://github.com/tektoncd/pipeline/pull/4041